libraries/botbuilder-azure/src/blobStorage.ts

```diff
@@ -117,14 +117,27 @@ export class BlobStorage implements Storage {
         return this.ensureContainerExists().then((container) => {
             return new Promise<StoreItems>((resolve, reject) => {
                 Promise.all<DocumentStoreItem>(sanitizedKeys.map((key) => {
-                    return new Promise((resolve, reject) => {
-                        this.client.getBlobMetadataAsync(container.name, key).then((blobMetadata) => {
-                            this.client.getBlobToTextAsync(blobMetadata.container, blobMetadata.name).then((result) => {
-                                let document: DocumentStoreItem = JSON.parse(result as any);
-                                document.document.eTag = blobMetadata.etag;
-                                resolve(document);
-                            }, err => resolve(null));
-                        }, (err) => resolve(null));
+                    return this.checkForBlob(container.name, key).then((blobResult) => {
+                        if (blobResult.exists) {
+                            return this.client.getBlobMetadataAsync(container.name, key).then((blobMetadata) => {
+                                return this.client.getBlobToTextAsync(blobMetadata.container, blobMetadata.name).then((result) => {
+                                    const document: DocumentStoreItem = JSON.parse(result as any);
+                                    return document;
```

Comment:
don't we still need the eTag?

Reply:
This is the document from blobStorage, so it already has an eTag in the current implementation.

There is a related issue saying that BotState should just be POCO (#297); so in the code below should we be bothering with eTags at all?

Reply:
The two (#297 and this) aren't mutually exclusive so I added the eTag semantics back to the PR. The PR is much simpler now than it was originally.